dali/test/python/test_operator_decoders_image.py

```diff
@@ -90,26 +88,125 @@ def log(img_type, size, device, threads):
-        super(DecoderPipelineFastIDC, self).__init__(batch_size, num_threads, 0, prefetch_queue_depth=1)
-        self.input = ops.readers.File(file_root = data_path,
-                                      shard_id = 0,
-                                      num_shards = 1)
-        self.decode = ops.decoders.Image(device = 'cpu', output_type = types.RGB, use_fast_idct=use_fast_idct)
+    anchor = fn.random.uniform(range=[0.05, 0.15], shape=(2,))
+    shape = fn.random.uniform(range=[0.5, 0.7], shape=(2,))
+    images_sliced_1 = fn.decoders.image_slice(jpegs,
+                                              anchor,
+                                              shape,
+                                              device = device,
+                                              hw_decoder_load = 0.7,
+                                              output_type = types.RGB,
+                                              axes = (0, 1))
 
-    def define_graph(self):
-        inputs, labels = self.input(name="Reader")
-        output = self.decode(inputs)
-        return (output, labels)
+    images = fn.decoders.image(jpegs,
+                               device = device,
+                               hw_decoder_load = 0.7,
+                               output_type = types.RGB)
+    images_sliced_2 = fn.slice(images,
+                               anchor,
+                               shape,
+                               axes = (0, 1))
+
+    return images_sliced_1, images_sliced_2
+
+# TODO(januszl): check padding behavior
+@pipeline_def
+def create_decoder_crop_pipeline(data_path, device):
+    jpegs, _ = fn.readers.file(file_root = data_path,
+                               shard_id = 0,
+                               num_shards = 1,
+                               name = "Reader")
+
+    crop_pos_x = fn.random.uniform(range=[0.1, 0.9])
+    crop_pos_y = fn.random.uniform(range=[0.1, 0.9])
+    w = random.randint(100, 250)
+    h = random.randint(100, 250)
+
+    images_crop_1 = fn.decoders.image_crop(jpegs,
+                                           device = device,
+                                           output_type = types.RGB,
+                                           hw_decoder_load = 0.7,
+                                           crop = (w, h),
+                                           crop_pos_x = crop_pos_x,
+                                           crop_pos_y = crop_pos_y)
+
+    images = fn.decoders.image(jpegs,
+                               device = device,
+                               hw_decoder_load = 0.7,
+                               output_type = types.RGB)
+
+    images_crop_2 = fn.crop(images,
+                            crop = (w, h),
+                            crop_pos_x = crop_pos_x,
+                            crop_pos_y = crop_pos_y)
+
+    return images_crop_1, images_crop_2
+
+@pipeline_def
+def create_decoder_random_crop_pipeline(data_path, device):
+    seed = 1234
```

Comment:
```suggestion
    seed = 1234
    random.seed(seed)
```
or use hard-coded w, h (which would probably be better).

Reply:
Done